src/add.cpp

```diff
@@ -236,7 +236,7 @@ RCP<const Basic> Add::from_dict(const RCP<const Number> &coef, umap_basic_num &&
         } else {
             insert(m, p->first, one);
             insert(m, p->second, one);
-            return rcp(new Mul(one, std::move(m)));
+            return rcp(new Mul(rcp_static_cast<const Integer>(one), std::move(m)));
```

Comment:
This should be `rcp_static_cast<const Number>` right? Otherwise it is first cast to an `Integer` and then to a `Number`